### PR TITLE
Merge/0.8.6

### PR DESCRIPTION
**Pull request by LlamaLad7**
Also pulls in 1 commit from the upcoming 0.8.7 since Mumfrey accidentally broke MixinExtras.
A couple of our changes were reverted to ease the merge of their upstream equivalents.
Generally I tried to reduce the diff where possible.
Works fine in the latest AOF7.

Requires changes in loader to support the new compatibility version.

### CODE REVIEW

File: src/main/java/org/spongepowered/asm/mixin/Mixins.java [hunk 1]
```diff
@@ -66,15 +67,25 @@ public final class Mixins {
...
     private Mixins() {}
     
+//    /**
+//     * Add multiple configurations
+//     * 
+//     * @param configFiles config resources to add
+//     */
+//    public static void addConfigurations(String... configFiles) {
+//        Mixins.addConfigurations(configFiles, null);
+//    }
```
[the remark this answers is not in the document]
Comment (LlamaLad7) [reply src/main/java/org/spongepowered/asm/mixin/Mixins.java]: No idea, seems reasonable enough to add it back.

**Comment by LlamaLad7** [comment]
I've asked Mumfrey to make one other change to support MixinExtras, will wait for his response before doing it here.

**Comment by LlamaLad7** [comment]
Pulled said changes in early here.

**Comment by LlamaLad7** [comment]
The fix to `NEW` descriptor filtering is potentially breaking so I will gate it.

**Review by modmuss50 — APPROVED** [review]
Basically impossible for me to review (due to the size and my lack of knowlage :D ). 

The compat fix looks good, im happy to merge and do a release of this so the loader side of things can be looked at.

**Pull request merged**